make-util.js

```diff
@@ -318,4 +320,115 @@ var getExternals = function (externals, destRoot) {
         });
     }
 }
-exports.getExternals = getExternals;
\ No newline at end of file
+exports.getExternals = getExternals;
+
+//------------------------------------------------------------------------------
+// task.json functions
+//------------------------------------------------------------------------------
+
+var createResjson = function (task, taskPath) {
+    var resources = {};
+    if (task.hasOwnProperty('friendlyName')) {
+        resources['loc.friendlyName'] = task.friendlyName;
+    }
+
+    if (task.hasOwnProperty('helpMarkDown')) {
+        resources['loc.helpMarkDown'] = task.helpMarkDown;
+    }
+
+    if (task.hasOwnProperty('description')) {
+        resources['loc.description'] = task.description;
+    }
+
+    if (task.hasOwnProperty('instanceNameFormat')) {
+        resources['loc.instanceNameFormat'] = task.instanceNameFormat;
+    }
+
+    if (task.hasOwnProperty('groups')) {
+        task.groups.forEach(function (group) {
+            if (group.hasOwnProperty('name')) {
+                resources['loc.group.displayName.' + group.name] = group.displayName;
+            }
+        });
+    }
+
+    if (task.hasOwnProperty('inputs')) {
+        task.inputs.forEach(function (input) {
+            if (input.hasOwnProperty('name')) {
+                resources['loc.input.label.' + input.name] = input.label;
+
+                if (input.hasOwnProperty('helpMarkDown')) {
+                    resources['loc.input.help.' + input.name] = input.helpMarkDown;
+                }
+            }
+        });
+    }
+
+    if (task.hasOwnProperty('messages')) {
+        Object.keys(task.messages).forEach(function (key) {
+            resources['loc.messages.' + key] = task.messages[key];
+        });
+    }
+
+    var resjsonPath = path.join(taskPath, 'Strings', 'resources.resjson', 'en-US', 'resources.resjson');
+    mkdir('-p', path.dirname(resjsonPath));
+    fs.writeFileSync(resjsonPath, JSON.stringify(resources, null, 2));
+};
+exports.createResjson = createResjson;
+
+var createTaskLocJson = function (taskPath) {
```

Comment:
I added separate functions for generating the task.loc.json vs the resources.resjson files. Common modules need to generate resources.resjson files, but not task.loc.json files.